File: 1357. Apply Discount Every n Orders/1357.go
package leetcode

type Cashier struct {
	myMap map[int]int
	n int
	people int
	discount int
}


func Constructor(n int, discount int, products []int, prices []int) Cashier {
	myMap := make(map[int]int,len(products))
	for i:=0; i<len(products); i++{
		myMap[products[i]] = prices[i]
	}
    return Cashier{
		myMap: myMap,
		n: n,
		people: 0,
		discount: discount,
	}
}


func (this *Cashier) GetBill(product []int, amount []int) float64 {
	this.people ++
	sum := 0
	if(this.people == this.n){
		this.people = 0
		for i:=0; i<len(product); i++{
			price, _ := this.myMap[product[i]]
			sum += (price*amount[i])
		}
		return float64(sum)*(1-float64(this.discount)/100)
	}else{
		for i:=0; i<len(product); i++{
			price, _ := this.myMap[product[i]]
			sum += (price*amount[i])
		}
		return float64(sum)
	}
}

File: 1357. Apply Discount Every n Orders/_1357.py
class Cashier:

    def __init__(self, n: int, discount: int, products: List[int], prices: List[int]):
        self.n = n
        self.discount = discount
        self.myDict = dict()
        for i in range(len(products)):
            self.myDict[products[i]] = prices[i]
        self.npeople = 0

    def getBill(self, product: List[int], amount: List[int]) -> float:
        self.npeople += 1
        if self.npeople == self.n:
            self.npeople = 0
            return sum([i[1]*self.myDict[i[0]] for i in zip(product, amount)]) *(1-self.discount/100)   
        else:
            return  sum([i[1]*self.myDict[i[0]] for i in zip(product, amount)])*1.0
        


# Your Cashier object will be instantiated and called as such:
# obj = Cashier(n, discount, products, prices)
# param_1 = obj.getBill(product,amount)